lib/client.js

```diff
@@ -737,68 +732,102 @@
     head.appendChild(script);
   };
 
-  HappnClient.prototype.getResources = function(callback) {
-
+  HappnClient.prototype.getResources = function (callback) {
     var _this = this;
 
-    if (typeof Primus == 'undefined') _this.getScript(_this.options.url + '/browser_primus.js', function(e) {
-      if (e) return callback(e);
-
-      if (typeof Primus == 'undefined') {
-
-        if (window && window.Primus) Primus = window.Primus;
-        else if (document && document.Primus) Primus = document.Primus;
-        else return callback(new Error('unable to fetch Primus library'));
+    if (typeof Primus == 'undefined')
+      _this.getScript(_this.options.url + '/browser_primus.js', function (e) {
+        if (e) return callback(e);
 
-        callback();
-      }
-    });
+        if (typeof Primus == 'undefined') {
+          if (window && window.Primus) Primus = window.Primus;
+          else if (document && document.Primus) Primus = document.Primus;
+          else return callback(new Error('unable to fetch Primus library'));
 
+          callback();
+        }
+      });
     else callback();
   };
 
-
-  HappnClient.prototype.stop = Promisify(function(callback) {
+  HappnClient.prototype.stop = Promisify(function (callback) {
     this.__connectionCleanup();
   });
 
-  HappnClient.prototype.__encryptLogin = function(parameters, publicKey) {
-
+  HappnClient.prototype.__encryptLogin = function (parameters, publicKey) {
     return {
-      encrypted: crypto.asymmetricEncrypt(publicKey, this.options.privateKey, JSON.stringify(parameters)),
+      encrypted: crypto.asymmetricEncrypt(
+        publicKey,
+        this.options.privateKey,
+        JSON.stringify(parameters)
+      ),
       publicKey: parameters.publicKey,
-      loginType: parameters.loginType != null ? parameters.loginType : 'password'
+      loginType:
+        parameters.loginType != null ? parameters.loginType : 'password'
     };
   };
 
-  HappnClient.prototype.__decryptLogin = function(loginResult) {
-
+  HappnClient.prototype.__decryptLogin = function (loginResult) {
     try {
-      return JSON.parse(crypto.asymmetricDecrypt(this.serverInfo.publicKey, this.options.privateKey, loginResult.encrypted));
+      return JSON.parse(
+        crypto.asymmetricDecrypt(
+          this.serverInfo.publicKey,
+          this.options.privateKey,
+          loginResult.encrypted
+        )
+      );
     } catch (e) {
       throw e;
     }
   };
 
-  HappnClient.prototype.__encryptPayload = function(message) {
+  HappnClient.prototype.__encryptPayload = function (message) {
+    var payload = '';
+    var self = this;
+
+    if (self.session.protocol === 'happn_4') {
```

Comment:
This is not necessary, will always be happn_4, the client would be synced with the latest protocol - if we leave this in here would cause issues when the protocol version is upped later